Multisig/UI/Settings/OwnerKeyManagement/GenerateOwnerKey/OnboardingGenerateKeyViewController.swift

```diff
@@ -73,14 +73,55 @@ class OnboardingGenerateKeyViewController: AddKeyOnboardingViewController {
     func startBackupFlow() {
         let backupController = BackupController(showIntro: true, seedPhrase: mnemonic)
         backupController.onComplete = { [weak self] in
-            self?.showKeyDetails()
+            guard let self = self else { return }
+            if self.canAddKeyAsOwner() {
+                self.startAddKeyAsOwner()
+            } else {
+                self.showKeyDetails()
+            }
         }
         backupController.onCancel = { [weak self] in
-            self?.showKeyDetails()
+            guard let self = self else { return }
+            self.dismiss(animated: true) {
+                if self.canAddKeyAsOwner() {
+                    self.startAddKeyAsOwner()
+                } else {
+                    self.showKeyDetails()
+                }
+            }
         }
         show(backupController, sender: self)
     }
 
+    func canAddKeyAsOwner() -> Bool {
+        guard let safe = try? Safe.getSelected() else { return false }
+        if !safe.isReadOnly {
+            return true
+        } else {
+            return true
+        }
+    }
+
+    func startAddKeyAsOwner() {
+        let addKeyAsOwnerController = AddKeyAsOwnerController(privateKey: privateKey)
+        addKeyAsOwnerController.onAdded = { [weak self] in
+            self?.dismiss(animated: true) {
+                self?.showKeyDetails()
+            }
+        }
```

Comment:
you can move the closure to a variable, and then assign this variable on all closures below.

Next question, if it doesn't matter which closure is called, should we have 3 closures? Maybe 1 completion block would be enough?

Reply:
Extracted closure to a variable, but I would keep 3 possible outcomes in case flow evolves in the future.